test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Models/Package/EmbeddedResourcesCapabilityTests.cs

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Moq;
+using NuGet.PackageManagement.UI.Models.Package;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+using NuGet.VisualStudio.Internal.Contracts;
+using Xunit;
+
+namespace NuGet.PackageManagement.UI.Test.Models
+{
+    public class EmbeddedResourcesCapabilityTests
+    {
+        [Fact]
+        public void LocalEmbeddedResourcesCapabilityCtor_SetsValues()
+        {
+            // Arrange
+            var icon = new Uri(@"C:\Path\To\Icon.png");
+            var license = new Uri(@"C:\Path\To\Icon.png");
+            var readme = new Uri(@"C:\Path\To\Icon.png");
+            var mockPackageFileService = new Mock<INuGetPackageFileService>();
+            var identity = new PackageIdentity("TestPackage", new NuGetVersion("1.0.0"));
+            var package = new TestPackageModel(identity);
+
+            // Act
+            var capability = new EmbeddedResourcesCapability(mockPackageFileService.Object, package, icon, license, readme);
+
+            // Assert
+            Assert.Equal(icon, capability.IconUri);
+            Assert.Equal(license, capability.LicenseUri);
+            Assert.Equal(readme, capability.ReadmeUri);
+        }
+
+        [Fact]
+        public void LocalEmbeddedResourcesCapabilityCtor_WithNullValues_SetsValues()
+        {
+            // Arrange
+            var mockPackageFileService = new Mock<INuGetPackageFileService>();
+            var identity = new PackageIdentity("TestPackage", new NuGetVersion("1.0.0"));
+            var package = new TestPackageModel(identity);
+
+            // Act
+            var capability = new EmbeddedResourcesCapability(mockPackageFileService.Object, package, null, null, null);
+
+            // Assert
+            Assert.Null(capability.IconUri);
+            Assert.Null(capability.LicenseUri);
+            Assert.Null(capability.ReadmeUri);
+        }
+
+        [Fact]
+        public void LocalEmbeddedResourcesCapabilityCtor_WithNullService_Throws()
```

Comment:
Consider a name like: Constructor_WithoutFileService_EnforcesRequiredParameter or Constructor_WithoutRequiredFileService_ThrowsException (Throws is fine too)

This adds more detail and describes what we're testing in the method name.